Integrations/integration-iDefense.yml

```diff
@@ -0,0 +1,411 @@
+
+        return {
+            Type : entryTypes.note,
+            Contents : results,
+            ContentsFormat : formats.json,
+            HumanReadable : tableToMarkdown('iDefense Reputations', merged_md),
+            EntryContext : mergeObjects([domain_threats.context, ip_threats.context, url_threats.context, {DBotScore : merged_scores}])
+        };
+
+    }
+
+    function get_threats(max_results) {
+        var cmd_url = 'threatindicator/v0/';
+        var results = get_full_data(cmd_url, {}, max_results);
+
+        return sort_threats(results);
+    }
+
+    function get_domain(domain) {
+        var threats_info = check_threats('Domain', domain, 'Name');
+        return {
+            Type : entryTypes.note,
+            Contents : threats_info.results,
+            ContentsFormat : formats.json,
+            HumanReadable : tableToMarkdown('iDefense Domain Reputation', threats_info.md),
+            EntryContext : mergeObjects([threats_info.context, {DBotScore : threats_info.scores}])
+        };
+    }
+
+    function get_ip(ip) {
+        if (!isValidIP(ip)) {
+            return {Type: entryTypes.error, Contents: 'IP - ' + ip + ' is not valid IP', ContentsFormat: formats.text};
+        }
+
+        var threats_info = check_threats('IP', ip, 'Address');
+        return {
+            Type : entryTypes.note,
+            Contents : threats_info.results,
+            ContentsFormat : formats.json,
+            HumanReadable : tableToMarkdown('iDefense IP Reputation', threats_info.md),
+            EntryContext : mergeObjects([threats_info.context, {DBotScore : threats_info.scores}])
+
+        };
+    }
+
+    function get_url(url) {
+        var threats_info = check_threats('URL', url, 'Data');
+        return {
+            Type : entryTypes.note,
+            Contents : threats_info.results,
+            ContentsFormat : formats.json,
+            HumanReadable : tableToMarkdown('iDefense URL Reputation', threats_info.md),
+            EntryContext : mergeObjects([threats_info.context, {DBotScore : threats_info.scores}])
+
+        };
+    }
+
+    function get_uuid(uuid) {
+        var cmd_url = 'threatindicator/v0/' + uuid;
+        var res = sendRequest('GET', cmd_url);
+
+        return sort_threats([res]);
+    }
+
+    // The command input arg holds the command sent from the user.
+    switch (command) {
+        case 'idefense-general':
+            // still having issues with this command.
+            return get_threats(args.max_result);
+        case 'domain':
+            return get_domain(args.domain);
+        case 'ip':
+            return get_ip(args.ip);
+        case 'url':
+            return get_url(args.url);
+        case 'uuid':
+            // still having issues with this command.
+            return get_uuid(args.uuid);
+        // This is the call made when pressing the integration test button.
+        case 'test-module':
+            return 'ok';
```

Comment:
unless I missed something, doesn't seem that you really check something here before returning ok